<launch>

  <arg name="image_path"	default="salt" />
  <arg name="vis"		default="false" />
  <arg name="board_ip" 		default="169.254.2.50" />
  <arg name="exposures" 	default="1" />
  <arg name="has_left_cam" 	default="true" />
  <arg name="has_right_cam" 	default="true" />
  <arg name="has_color_cam" 	default="true" />
  <arg name="stereo_cam_topic"  default="vrm3dvision" />
  <arg name="pc_color_option"   default="none" />
  <arg name="remove_outliers"   default="true" />
  <arg name="remove_dominant_plane"   	default="true" />
  <arg name="certainty_type"	default="0" />

  <arg name="layers"		default="8" />

  <arg name="point_cloud_topic" default="point_cloud" />
  <arg name="point_cloud_trimmed_topic" default="point_cloud_trimmed" />

        <group ns="/vrm3dvision/">

                <node pkg="vrm3dvision" type="vrm_log_node" name="VrmLogNode" output="screen">
			<param name="vrm_board_ip" value="$(arg board_ip)" />
                </node>

                <node pkg="vrm3dvision" type="vrm_command_node" name="VrmCommandNode" output="screen">
			<param name="initial_mode" value="11"/>
			<param name="initial_expo_grey" type="string" value="2.0" /> 
			<param name="vrm_board_ip" value="$(arg board_ip)" />
                </node>

               <node pkg="vrm3dvision" type="vrm_image_proc" name="VrmImageProcNode" output="screen">
			<param name="vrm_board_ip" value="$(arg board_ip)" />
			<param name="has_left_camera" value="true" />
			<param name="has_right_camera" value="true" />
			<param name="has_color_camera" value="true" />
			<param name="stereo_camera_topic" value="vrm_stereo_camera" />
			<param name="color_cam_topic" value="color_cam" />
			<param name="visualization" value="$(arg vis)" />
			<param name="no_of_layers_structured_light" value="$(arg layers)" />
			<param name="point_cloud_pub_topic" value="$(arg point_cloud_topic)" />			
			<param name="point_cloud_trimmed_pub_topic" value="$(arg point_cloud_trimmed_topic)" />
			<param name="publish_trimmed_point_cloud" value="true" />			
			<param name="z_trim_distance" value="0.40" />			
			<!--<param name="calibration_path" value="$(arg image_path)" /> -->
			<param name="pc_color_option" value="$(arg pc_color_option)" />
 			<param name="remove_outliers" value="$(arg remove_outliers)" />
			<param name="remove_dominant_plane" value="$(arg remove_dominant_plane)" />	
 			<param name="mean_k" value="20" />
 			<param name="std_dev_thresh" value="1.5" />
			<param name="certainty_type" value="$(arg certainty_type)" />
			<param name="allow_robot_control" value="false" />
                </node>		
		<node pkg="vrm3dvision" type="pcl_pointcloud_viewer" name="PCLPCviewer" output="screen">
			<param name="point_cloud_subscriber_topic" value="$(arg point_cloud_topic)" />
		</node> 
		<node pkg="vrm3dvision" type="pcl_pointcloud_viewer" name="PCLPCviewer2" output="screen">
			<param name="point_cloud_subscriber_topic" value="$(arg point_cloud_trimmed_topic)" />
		</node> 

		<node pkg="rqt_service_caller" type="rqt_service_caller" name="RqtServiceCaller" />

        </group>

	<node pkg="tf" type="static_transform_publisher" name="object_to_grasp"
	    args="0.005 -0.004 0.294 0 1.570796327 -2.35619449  /object /object_grasp 100" />

	<node pkg="tf" type="static_transform_publisher" name="grasp_to_pregrasp"
	    args="-0.10 0.00 0.00 0 0 0  /object_grasp /object_pre_grasp 100" />

	<node pkg="tf" type="static_transform_publisher" name="flange_to_tcp"
	    args="0.26 0.00 0.00 0 0 0  /ee_link /tcp_link 100" />
</launch>
